plugins/device/deviceutils/store-utils.go

```diff
@@ -174,3 +178,77 @@ func IsVgExist(vgname string) bool {
 	}
 	return false
 }
+
+// getDeviceAvailableSize gets the device size and vgName using device Path
+func getDeviceAvailableSize(devicePath string) (string, uint64, error) {
+	devices, err := GetDevices()
+	if err != nil {
+		return "", 0, err
+	}
+	deviceVgName := strings.Split(devicePath, "/")[2]
+	for _, d := range devices {
+		if d.VgName == deviceVgName {
+			return d.VgName, d.AvailableSize, nil
+		}
+	}
+	return "", 0, nil
+}
+
+// CheckForAvailableVgSize prepares a brickName to vgName mapping in order to use while expanding lvm
+// Also it checks for sufficient space available on devices of current node.
+func CheckForAvailableVgSize(expansionSize uint64, bricksInfo []brick.Brickinfo) (map[string]string, bool, error) {
+
+	// map of brickName to vgName. Used while extending lv
+	brickVgMapping := make(map[string]string)
+
+	// map of deviceName to required free size in that device
+	requiredDeviceSizeMap := make(map[string]uint64)
+
+	// Map device Name to required free space on the device
```

Comment:
map deviceName to required free space on the device

Reply:
Done